src/EFCore.SqlServer/Infrastructure/Internal/SqlServerModelValidator.cs

```diff
@@ -209,6 +210,105 @@ public override void Validate(IModel model, IDiagnosticsLogger<DbLoggerCategory.
             }
         }
 
+        /// <summary>
+        ///     This is an internal API that supports the Entity Framework Core infrastructure and not subject to
+        ///     the same compatibility standards as public APIs. It may be changed or removed without notice in
+        ///     any release. You should only use it directly in your code with extreme caution and knowing that
+        ///     doing so can result in application failures when updating to a new Entity Framework Core release.
+        /// </summary>
+        protected virtual void ValidateTemporal(
+            IModel model,
+            IDiagnosticsLogger<DbLoggerCategory.Model.Validation> logger)
+        {
+            foreach (var temporalEntityType in model.GetEntityTypes().Where(t => t.IsTemporal()))
+            {
+                var entityTypeName = temporalEntityType.ShortName();
+
+                if (temporalEntityType.BaseType != null)
+                {
+                    throw new InvalidOperationException(SqlServerStrings.TemporalOnlyOnRoot(entityTypeName));
+                }
+
+                if (temporalEntityType.TemporalHistoryTableName() == null)
+                {
+                    throw new InvalidOperationException(SqlServerStrings.TemporalNeedsHistoryTable(entityTypeName));
+                }
+
+                ValidateTemporalPeriodProperty(temporalEntityType, periodStart: true);
+                ValidateTemporalPeriodProperty(temporalEntityType, periodStart: false);
+
+                if (temporalEntityType.GetDerivedTypes().Any())
+                {
+                    var tableMappings = temporalEntityType.GetDerivedTypes().Select(t => t.GetTableName()).Distinct();
```

Comment:
```suggestion
                    var tableMappings = temporalEntityType.GetDerivedTypes().Select(t => t.GetTableName()).Distinct().ToList();
```

You can use this value in the condition above.